File: Server/src/main/content/global/interaction/npc/CowNPC.java
package content.global.interaction.npc;

import core.game.node.entity.npc.AbstractNPC;
import core.game.world.map.Location;
import core.plugin.Initializable;
import core.utilities.RandomFunction;

/**
 * The Cow npc.
 */
@Initializable
public final class CowNPC extends AbstractNPC {

    private static final String FORCE_CHAT = "Moo";

    /**
     * Instantiates a new Cow npc.
     */
    public CowNPC() {
        super(0, null, true);
    }

    /**
     * Instantiates a new Cow npc.
     *
     * @param id       the id
     * @param location the location
     */
    public CowNPC(int id, Location location) {
        super(id, location, true);
    }

    @Override
    public void tick() {
        if (RandomFunction.random(45) == 5 && this.getId() != 1766) { // calves don't moo
            sendChat(FORCE_CHAT);
        }
        super.tick();
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new CowNPC(id, location);
    }

    @Override
    public int[] getIds() {
        return new int[]{81, 397, 955, 1766, 1767, 3309};
    }

}

File: Emulator/src/main/content/global/handlers/npc/CowNPC.kt
package content.global.handlers.npc

import org.rs.consts.NPCs
import core.api.sendChat
import core.game.node.entity.npc.NPC
import core.game.node.entity.npc.NPCBehavior
import core.tools.RandomFunction

/**
 * Represents the Cow NPC.
 */
class CowNPC : NPCBehavior(NPCs.COW_81, NPCs.COW_397, NPCs.COW_955, NPCs.COW_CALF_1766, NPCs.COW_1767, NPCs.COW_3309) {

    override fun tick(self: NPC): Boolean {
        if (RandomFunction.random(45) == 5 && self.id != NPCs.COW_CALF_1766) {
            sendChat(self, "Moo")
        }
        return true
    }

}
